WordPress/Classes/System/WordPressAppDelegate.swift

```diff
@@ -56,6 +56,7 @@ class WordPressAppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, willFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
         window = UIWindow(frame: UIScreen.main.bounds)
 
+        AccountService.init(managedObjectContext: ContextManager.shared.mainContext).mergeDuplicatesIfNecessary()
```

Comment:
Instead of doing this on launch, what do you think about putting it right before syncBlogs? That way it's closer to the other de-duplication code, and we can trigger it via pull-to-refresh.

Reply:
Dunno. It shouldn't need to be called so frequently and if there are multiple accounts then there is a slight overhead calling it. I'd prefer to not call it aggressively but if you have a strong feeling about it we could change it.

Reply:
I can't make up my mind, I see benefits for both cases. If it stays here though, I'd remove from the initialization critical path. Maybe put it async on a background context like we do others:

```swift
// Deferred tasks to speed up app launch
DispatchQueue.global(qos: .background).async { [weak self] in
    MediaCoordinator.shared.refreshMediaStatus()
    PostCoordinator.shared.refreshPostStatus()
    MediaFileManager.clearUnusedMediaUploadFiles(onCompletion: nil, onError: nil)
    self?.uploadsManager.resume()
}
```

And you can drop the `.init` 😁

Reply:
> If it stays here though, I'd remove from the initialization critical path. 

Hmm.  I think we'd want to run this before the authtoken fixer tho.

Reply:
I'm not sure I fully understand this scenario, but what I think you're saying is that:

1. There is a duplicate account
2. The auth token is missing
3. The token fixer asks the user to re-authenticate
4. If the user cancels, the default account is removed
5. If the de-duplication runs after that it would promote the duplicate as the default account but would still have no token

If that's the case, since #11245 we would just prompt for re-authentication again on the first network request that failed, so I'm not too worried about that.

Reply:
Fair point.